02-cpu-configure/pcie-remove.c

```diff
@@ -0,0 +1,56 @@
+#include <fcntl.h>
+#include <unistd.h>
+#include <stdio.h>
+#include <errno.h>
+#include <string.h>
+
+int main(int argc, char** argv) {
+  if (argc != 2) {
+    fprintf(stderr, "Usage: %s slot_num\n", argv[0]);
+    return 1;
+  }
+
+  char slot_addr_path[256];
+  snprintf(slot_addr_path, 256, "/sys/bus/pci/slots/%s/address", argv[1]);
+  int fd_addr = open(slot_addr_path, O_RDONLY);
+  if (fd_addr < 0) {
+    fprintf(stderr, "Slot %s not found!\n", argv[1]);
+    return 2;
+  }
+
+  char slot_addr[64];
+  ssize_t bytes = read(fd_addr, slot_addr, 64);
```

Comment:
I'd suggest using `sizeof` for all buffer sizes (in read and snprintf calls) instead of hardcoding sizes in two places.